### PR TITLE
feat: surface HTTP connection manager metrics

### PR DESCRIPTION
**Pull request by ianbotsf**
*Issue #, if available:*

https://github.com/awslabs/smithy-kotlin/issues/893

*Description of changes:*

This change makes HTTP manager metrics available in the Kotlin API.

**Companion PR**: https://github.com/awslabs/smithy-kotlin/pull/1017

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by lauzadis — APPROVED** [review]

File: aws-crt-kotlin/common/src/aws/sdk/kotlin/crt/http/HttpClientConnectionManager.kt [hunk 1]
```diff
@@ -14,6 +14,11 @@ import aws.sdk.kotlin.crt.Closeable
 public expect class HttpClientConnectionManager(options: HttpClientConnectionManagerOptions) :
     Closeable,
     AsyncShutdown {
+    /**
+     * The active metrics for this connection manager
+     */
+    public val managerMetrics: HttpManagerMetrics
```
Comment (aajtodd) [inline aws-crt-kotlin/common/src/aws/sdk/kotlin/crt/http/HttpClientConnectionManager.kt:20]: **nit**: Wouldn't `metrics` suffice here given it's namespaced under `HttpClientConnectionManager`

File: aws-crt-kotlin/common/src/aws/sdk/kotlin/crt/http/HttpManagerMetrics.kt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.sdk.kotlin.crt.http
+
+public data class HttpManagerMetrics(
```
Comment (aajtodd) [inline aws-crt-kotlin/common/src/aws/sdk/kotlin/crt/http/HttpManagerMetrics.kt:7]: **question**: Are these connection manager metrics? If so `HttpManager` seems ambiguous and should probably be `HttpConnectionManagerMetrics` or `HttpConnectionMetrics`

**fix**: Missing docs on these metrics. I know they are meant to match SRA but probably should document them still and any differences/quirks that may lie herein.

**Review by aajtodd — APPROVED** [review]
Couple questions/nit but looks good.

File: aws-crt-kotlin/common/src/aws/sdk/kotlin/crt/http/HttpClientConnectionManager.kt (hunk 1 above)
Comment (ianbotsf) [reply aws-crt-kotlin/common/src/aws/sdk/kotlin/crt/http/HttpClientConnectionManager.kt:20]: Similar to other Java→Kotlin mapping classes, I chose to keep this [the same as the underlying property](https://github.com/awslabs/aws-crt-java/blob/3eea28d5f617f408655054efd33278ce84d0563b/src/main/java/software/amazon/awssdk/crt/http/HttpClientConnectionManager.java#L219-L227). If you feel strongly I can change it to be simpler (which I would generally prefer when not mapping existing classes).

File: aws-crt-kotlin/common/src/aws/sdk/kotlin/crt/http/HttpManagerMetrics.kt (hunk 2 above)
Comment (ianbotsf) [reply aws-crt-kotlin/common/src/aws/sdk/kotlin/crt/http/HttpManagerMetrics.kt:7]: Similar to other Java→Kotlin mapping classes, I chose to keep this [the same as the underlying class](https://github.com/awslabs/aws-crt-java/blob/3eea28d5f617f408655054efd33278ce84d0563b/src/main/java/software/amazon/awssdk/crt/http/HttpManagerMetrics.java#L3). If you feel strongly I can change it to be clearer (which I would generally prefer when not mapping existing classes).

**Pull request merged**